platforms/documentation/docs/src/docs/userguide/dep-man/dependency-management/resolution_rules.adoc

```diff
@@ -0,0 +1,474 @@
+include::sample[dir="snippets/dependencyManagement/customizingResolution-classifierSubstitutionRule/groovy",files="consumer/build.gradle[tags=substitution_rule]"]
+====
+
+This rule effectively replaces any dependency on `quasar-core` found in the graph with a dependency without classifier.
+
+If you need to substitute with a specific classifier or artifact, you can specify the classifier or artifact details in the substitution rule.
+
+For more detailed information, refer to:
+
+- Artifact selection via the link:{javadocPath}/org/gradle/api/artifacts/DependencySubstitutions.Substitution.html[Substitution DSL]
+- Artifact selection via the link:{javadocPath}/org/gradle/api/artifacts/DependencySubstitution.html#artifactSelection-org.gradle.api.Action-[DependencySubstitution API]
+- Artifact selection via the link:{javadocPath}/org/gradle/api/artifacts/ResolutionStrategy.html#dependencySubstitution-org.gradle.api.Action-[ResolutionStrategy API]
+
+
+[[sec:component_selection_rules]]
+== Using component selection rules
+
+Component selection rules may influence which component instance should be selected when multiple versions are available that match a version selector.
+Rules are applied against every available version and allow the version to be explicitly rejected.
+
+This allows Gradle to ignore any component instance that does not satisfy conditions set by the rule.
+Examples include:
+
+* For a dynamic version like `1.+` certain versions may be explicitly rejected from selection.
+* For a static version like `1.4` an instance may be rejected based on extra component metadata such as the Ivy branch attribute, allowing an instance from a subsequent repository to be used.
+
+Rules are configured via the link:{groovyDslPath}/org.gradle.api.artifacts.ComponentSelectionRules.html[ComponentSelectionRules] object.
+Each rule configured will be called with a link:{groovyDslPath}/org.gradle.api.artifacts.ComponentSelection.html[ComponentSelection] object as an argument that contains information about the candidate version being considered.
+Calling link:{groovyDslPath}/org.gradle.api.artifacts.ComponentSelection.html#org.gradle.api.artifacts.ComponentSelection:reject(java.lang.String)[ComponentSelection.reject(java.lang.String)] causes the given candidate version to be explicitly rejected, in which case the candidate will not be considered for the selector.
+
+The following example shows a rule that disallows a particular version of a module but allows the dynamic version to choose the next best candidate:
+
+====
+include::sample[dir="snippets/dependencyManagement/customizingResolution-selectionRule/kotlin",files="build.gradle.kts[tags=reject-version-1-1]"]
+include::sample[dir="snippets/dependencyManagement/customizingResolution-selectionRule/groovy",files="build.gradle[tags=reject-version-1-1]"]
+====
```

Comment:
🤔 Creating a `rejectConfig` configuration here: 1) is a legacy configuration that is no longer encouraged - we should create it with a role and 2) is needlessly confusing as it introduces the string syntax to reference it in `dependencies` - this example should work fine using `implementation`

Reply:
Fixed.